docs-src/docs/guides/terraform_beyond_layer0.md

```diff
@@ -0,0 +1,519 @@
+
+### Locking
+Not all remote backends support locking (locking ensures only one person is able to change the state at a time). The `S3` backend we used earlier in the example supports locking which is disabled by default. The `S3` backend uses a DynamoDB table to acquire a lock before making a change to the state file. To enable locking, you need to specify `locking_table` property with the name of an existing DynamoDB table. The DynamoDB table also needs primary key named `LockID` of type `String`.
+
+### Security
+A Terraform state file is written in plain text. This can lead to a situation where deploying resources that require sensitive data can result in the sensitive data being stored in the state file. To minimize exposure of sensitive data, you can enable [server side encryption](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html) of the state file by adding property `encrypt` set to `true`.
+
+This will ensure that the file is encrypted in S3 and by using a remote backend, you will also have the added benefit of the state file not being persisted to disk locally as it will only ever be held in memory by Terraform.
+
+For securing the state file further, you can also enable access logging on the S3 bucket you are using for the remote backend, which can help track down invalid access should it occur.
+
+## Part 6: Terraform Configuration Structure
+
+There are many different approaches to setup your Terraform code structure. Although there is no one prescrived approach, whatever approach you take needs to be catered for the needs of your particular project. Keeping that in mind; the file structure for [Terraform beyond Layer0 example](https://github.com/quintilesims/layer0-examples/blob/master/terraform-beyond-layer0/example-1) is as below:
+
+```
+example1/
+  ─ main.tf  
+  ─ variables.tf  
+  ─ output.tf  
+  + modules/  
+      + guestbook_service/
+        ─ main.tf  
+        ─ variables.tf  
+        ─ output.tf
+      + service2/
+      + service3/
+```
+
+Here we are making use of Terraform [Modules](https://www.terraform.io/docs/modules/index.html). Modules in Terraform are self-contained packages of Terraform configurations, that are managed as a group. Modules are used to create reusable components in Terraform as well as for basic code organization. In this example, we are using modules to separate each service and making it consumable as a module.
+
+If you wanted to add a new service, you can create new folder service folder inside the ./modules. If you wanted to you could even run multiple copies of the same service. See here for more information about [Creating Modules](https://www.terraform.io/docs/modules/create.html).
+
+When creating a module, ensure that resources you are creating are prefixed with the environment and the module's name variable to ensure your resources are unique for each layer0 environment and each reference to a module.
```

Comment:
Ok - I think was over compensating for an impractical use case here. For example, if you wanted to for some crazy reason scale your service like below, then you would need prefixed names for resources? I just wasn't sure if doing something like below would make sense in any scenario.

```
module "guestbook" {
  source = "./modules/guestbook_service"

  name                    = "guestbook1"
  ...
}

module "guestbook2" {
  source = "./modules/guestbook_service"

  name                    = "guestbook2"
  ...
}
```